src/components/layout/page-header/PageHeader.tsx

```diff
@@ -19,17 +19,48 @@
  */
 
 import styled from '@emotion/styled';
-import { forwardRef, ReactNode } from 'react';
+import { CSSProperties, forwardRef, ReactNode } from 'react';
+import { useForwardedRef } from '~common/helpers/useForwardedRef';
+import { useResizeObserver } from '~common/helpers/useResizeObserver';
 import { cssVar } from '~utils/design-tokens';
 import { Divider } from '../../divider';
-import { PageGridArea } from '../LayoutTypes';
+import { PageGridArea, PageHeaderScrollBehavior } from '../LayoutTypes';
+
+enum PageHeaderArea {
+  breadcrumbs = 'breadcrumbs',
+  main = 'main',
+  actions = 'actions',
+  nav = 'nav',
+  divider = 'divider',
+}
+
+const PageHeaderBehaviorStyles: Record<PageHeaderScrollBehavior, CSSProperties> = {
```

Comment:
I think it should either start with a lowercase or be full uppercase like a constant.

Also maybe let's move all this logic at the bottom of the file? The main actor of this file is the PageHeaderRoot component, right now it's lost in the middle.